feat: add comprehensive validation for GameParams during game initial…

manoahLinks opened the pull request:
…ization

- Implemented strict validation of all GameParams fields in the game initialization logic.
- Prevents invalid or disruptive parameters (e.g., zero/negative blinds, player count out of bounds, etc.).
- Ensures only sensible game settings are accepted, improving game flow and reliability.

Closes #179

## Description   
<!-- Provide a brief summary of the changes in this PR. Explain what was modified and why. -->    

## Related Issues   
<!-- Link any related issues using `Closes #issue_number` or `Fixes #issue_number`. This helps track bugs and feature requests. -->    

## Changes Made   - [ ] 
<!-- List key changes made in this PR. Use bullet points for clarity. -->    

## How to Test  
<!-- Explain how a reviewer can test these changes. Provide commands, steps, or expected results if applicable. -->    
 
## Screenshots (if applicable)  
<!-- If your PR affects the UI, upload screenshots to show the changes visually. -->    
  
## Checklist  
- [ ] My code follows the project's coding style.  
- [ ] I have tested these changes locally.   
- [ ] Documentation has been updated where necessary.

manoahLinks committed:
feat: add comprehensive validation for GameParams during game initialization

- Implemented strict validation of all GameParams fields in the game initialization logic.
- Prevents invalid or disruptive parameters (e.g., zero/negative blinds, player count out of bounds, etc.).
- Ensures only sensible game settings are accepted, improving game flow and reliability.

Closes #179

Birdmannn commented on poker-texas-hold-em/contract/src/traits/game.cairo:
```diff
@@ -16,6 +16,14 @@ pub impl GameImpl of GameTrait {
                 // Validate custom params
                 assert(params.max_no_of_players > 1, GameErrors::MIN_PLAYER);
                 assert(params.big_blind > params.small_blind, GameErrors::INVALID_BLIND_PLAYER);
+                assert(params.small_blind > 0, GameErrors::INVALID_GAME_PARAMS);
+                assert(params.big_blind > 0, GameErrors::INVALID_GAME_PARAMS);
+                assert(params.no_of_decks > 0, GameErrors::INVALID_GAME_PARAMS);
+                assert(params.min_amount_of_chips > 0, GameErrors::INVALID_GAME_PARAMS);
+                assert(params.blind_spacing > 0, GameErrors::INVALID_GAME_PARAMS);
```
We need the adequate minimum stats set as a constant using your gaming IQ. So using the minimum as zero is not quite good in real life scenarios... I don't know if you understand me sir.

manoahLinks replied:
please could you shed more lights on this

Birdmannn replied:
For example, a small blind means the minimum amount if betting chips someone can use to start a game round. For now, we take 10 chips to 1 usd. For an "adequate minimum stats", I should know just observe within myself that the minimum value should atleast be around 20usd, that'll be equals 200 chips. So the assert statement would be something like
```cairo
assert(params.small_blind >= 200, ...
```
And I would also want to put a cap on the max value of the small_blind too... and based on your research on how it happens in real life, you can put the max cap. 

Note that this is just an example, but you can preserve the exchange rate for now.

manoahLinks replied:
done, please review the recent changes

Birdmannn reviewed (commented):
Check this out

manoahLinks committed:
feat: enforce realistic minimum and maximum values for GameParams

- Defined constants for minimum and maximum values of key game parameters (players, blinds, decks, chips, blind spacing) based on real-world poker standards.
- Updated game initialization validation logic to use these constants, ensuring only sensible and playable game settings are accepted.
- Improved code readability and maintainability by centralizing parameter limits.

Birdmannn reviewed (commented):
You tried
Please @manoahLinks fetch from upstream and ensure `sozo test` passes

manoahLinks committed:
Merge branch 'v1' into doctor_needed

manoahLinks commented:
done ser, sozo test passes and now ready for final review
